File: C++/Algorithms/String/reverseString.cpp
#include <cstring>

/**
 * Reverses the given string
 * @param char *string
 * @return int
 */
char * reverseString(char *string)
{
    int length = std::strlen(string);
    char *reverseString;

    for (int i = length - 1, j = 0;i > -1;i--, j++) {
        reverseString[j] = string[i];
    }

    return reverseString;
}

File: PHP/Algorithms/String/reverseString.php
<?php

/**
 * Reverses the given string
 *
 * @param string $string
 * @return string
 */
function reverseString($string)
{
    $reverseString = '';
    $length = strlen($string);

    for ($i = $length - 1;$i > -1;$i--) {
        $reverseString .= $string[$i];
    }

    return $reverseString;
}
